pulsar-client/src/main/java/org/apache/pulsar/client/impl/metrics/InstrumentProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.client.impl.metrics;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.Optional;
+import org.apache.pulsar.PulsarVersion;
+import org.apache.pulsar.client.impl.conf.ClientConfigurationData;
+
+public class InstrumentProvider {
+
+    public static final InstrumentProvider NOOP = new InstrumentProvider(OpenTelemetry.noop());
+
+    private final Meter meter;
+
+    public InstrumentProvider(OpenTelemetry otel) {
+        if (otel == null) {
+            // By default, metrics are disabled, unless the OTel java agent is configured.
+            // This allows to enable metrics without any code change.
```

Comment:
I doubt about this, since OTEL agent's lib and pulsar client's lib might be loaded by different classloaders, even through there is a javaagent installed on the app, but GlobalOpenTelemetry.get() may returns noop.
Maybe I'm wrong, but it might be happens.

Reply:
There is always the possibility of weird issues with different classloaders. In general terms, though, it does work fine that if you enable the OTel agent, you will get Pulsar client metrics without passing the OpenTelemetry object when building the `PulsarClient` instance.